### PR TITLE
[Refactor] Add error message in log, when schema change call add_chunk.

### PR DESCRIPTION
**Pull request by Linkerist**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

File: be/src/storage/schema_change.cpp [hunk 1]
```diff
@@ -849,11 +849,11 @@ bool SchemaChangeDirectly::process(TabletReader* reader, RowsetWriter* new_rowse
...
             return false;
         }
-        if (auto tmp_st = new_rowset_writer->add_chunk(*new_chunk); !tmp_st.ok()) {
+        if (auto st = new_rowset_writer->add_chunk(*new_chunk); !st.ok()) {
             std::string err_msg = Substitute(
                     "failed to execute schema change. base tablet:$0, new_tablet:$1. err msg: failed to add chunk to "
-                    "rowset writer",
-                    base_tablet->tablet_id(), new_tablet->tablet_id());
+                    "rowset writer: $2",
+                    base_tablet->tablet_id(), new_tablet->tablet_id(), st.to_string());
```
Comment (Linkerist) [inline be/src/storage/schema_change.cpp:856]: or can be st.get_error_msg, the output will be shorter without call stack.

st.to_string():
```
W0704 19:53:12.844352 161050 schema_change.cpp:1529] failed to process the schema change. from tablet 10007.808453491.ad4aa83d2cfce452-5c8c977a33fe39a9 to tablet 10012.481463293.2b43120c99541c59-b712c8e2740efaa0 version=0-2 error Internal error: failed to execute schema change. base tablet:10007, new_tablet:10012. err msg: failed to add chunk to rowset writer: Invalid argument: string length(8) > limit(5)
../src/storage/rowset/column_writer.cpp:901 check_string_lengths(column)
../src/storage/rowset/segment_writer.cpp:295 _column_writers[i]->append(*col)
../src/storage/rowset/beta_rowset_writer.cpp:259 _segment_writer->append_chunk(chunk)
```
st.get_error_msg():
```
W0704 19:48:20.816605 143400 schema_change.cpp:1529] failed to process the schema change. from tablet 10007.808453491.234ef3e8be7bd08f-b9a25c11558875bf to tablet 10012.685598212.3040bf248e36d895-6bfd24292b7e948d version=0-2 error Internal error: failed to execute schema change. base tablet:10007, new_tablet:10012. err msg: failed to add chunk to rowset writer: string length(8) > limit(5)
```

**Review by trueeyu — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: be/src/storage/schema_change.cpp [hunk 2]
```diff
@@ -927,11 +927,11 @@ Status SchemaChangeDirectly::processV2(TabletReader* reader, RowsetWriter* new_r
...
         ChunkHelper::padding_char_columns(char_field_indexes, new_schema, new_tablet->tablet_schema(), new_chunk.get());
 
-        if (auto tmp_st = new_rowset_writer->add_chunk(*new_chunk); !tmp_st.ok()) {
+        if (auto st = new_rowset_writer->add_chunk(*new_chunk); !st.ok()) {
             std::string err_msg = Substitute(
                     "failed to execute schema change. base tablet:$0, new_tablet:$1. err msg: failed to add chunk to "
-                    "rowset writer",
-                    base_tablet->tablet_id(), new_tablet->tablet_id());
+                    "rowset writer: $2",
+                    base_tablet->tablet_id(), new_tablet->tablet_id(), st.to_string());
```
[the remark this answers is not in the document]
Comment (Linkerist) [reply be/src/storage/schema_change.cpp]: changed in https://github.com/StarRocks/starrocks/pull/8246/commits/a8a2d8545c852f7970bae69843c5d5216bdc5d70

**Review by decster — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Pull request merged**